### DISCUSSION TITLE
[@types/jsonwebtoken] Add default export statement

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [DefinitelyTyped/DefinitelyTyped#62137] [Issues with a @types package] [answered]
The `jsonwebtoken` package's root `index.js` file exports it's methods as

```js
module.exports = {
  decode: require('./decode'),
  verify: require('./verify'),
  sign: require('./sign'),
  JsonWebTokenError: require('./lib/JsonWebTokenError'),
  NotBeforeError: require('./lib/NotBeforeError'),
  TokenExpiredError: require('./lib/TokenExpiredError'),
};
```

The `@types/jsonwebtoken` package (correctly) types these as 

```ts
export function decode(...)
export function verify(...)
export function sign(...)
...etc
```

However, in an ESM context, that results in an error:

```js
import jwt from 'jsonwebtoken';
// Module '"/path/to/@types/jsonwebtoken/index"' has no default export
```

Seeing the package only has a single (default) CJS export, this can't be circumvented in ESM with named imports[^1], or `* as x` imports[^2].

This can be resolved in the type definitions by adding the following to the end of the declaration file:

```ts
declare const jsonwebtoken = { decode, verify, sign, ...etc }
export default jsonwebtoken;
```

Is that an acceptable change? Happy to open a PR if accepted 🙂

[^1]: `SyntaxError: The requested module 'jsonwebtoken' does not provide an export named 'sign'`
[^2]: `TypeError: jwt.sign is not a function`

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment, marked as answer]
the error is proper, check if your tsconfig supports syntactic sugar with `esModuleInterop`, which allows to use syntactic default imports:
https://www.typescriptlang.org/tsconfig#esModuleInterop
IMO this is use-case TS config issue, no action expected for declaration file (which is correct, as you noted)

**Reply by rijkvanzanten** [reply]
Gotcha. I was under the impression that `esModuleInterop` would generate a bunch of cruft in the output JS (as it used to with CJS), but when compiling to ESM that doesn't seem to be the case. Setting `esModuleInterop` seems to be an acceptable solution 👍🏻
